src/sentry/nodestore/bigtable/backend.py

```diff
@@ -25,6 +27,32 @@
 json_loads = _default_decoder.decode
 
 
+_connection_lock = Lock()
+_connection_cache = {}
+
+
+def get_connection(project, instance, table, options):
+    key = (project, instance, table)
+    try:
+        # Fast check for an existing connection cached
+        return _connection_cache[key]
+    except KeyError:
+        # if missing, we acquire our lock to initialize a new one
+        with _connection_lock:
+            try:
+                # It's possible that the lock was blocked waiting
+                # on someone else who already initialized, so
+                # we first check again to make sure this isn't the case.
+                return _connection_cache[key]
+            except KeyError:
+                _connection_cache[key] = (
```

Comment:
I'm assuming you looked to make sure this is actually threadsafe. (I didn't.)

Reply:
Yeah, it is.